Assets/GRPC.NET/Scripts/GRPCBestHttpHandler.cs

```diff
@@ -144,20 +131,68 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage grpcRe
                 isHeader = false;
             };
 
-
-            // For each incoming DATA package we write data trough to gRPC
-            bestRequest.OnStreamingData += (_, _, fragment, length) =>
+            // For each incoming DATA package we write data through to gRPC
+            bestRequest.DownloadSettings.OnDownloadStarted += (_, _, stream) =>
             {
-                // Write incoming DATA package and immediately flush
-                incomingDataStream.Write(fragment, 0, length);
-                incomingDataStream.Flush();
-                return true;
+                try
```

Comment:
The new download streaming pattern:
1) drain all available download at the time of this callback
2) if the stream isn't done, kick off a Task and poll for new data until complete. The new "detatched" pattern will hold resources open until we finish draining and close the stream